WalletWasabi.Tests/UnitTests/WabiSabi/Integration/WabiSabiHttpApiIntegrationTests.cs

```diff
@@ -42,6 +46,78 @@ public async Task RegisterSpentOrInNonExistentCoinAsync()
 			Assert.Equal(WabiSabiProtocolErrorCode.InputSpent, wex.ErrorCode);
 		}
 
+		[Fact]
+		public async Task SoloCoinJoinTest()
+		{
+			const int InputCount = 2;
+
+			// At the end of the test a coinjoin transaction has to be created and broadcasted.
+			// we wait for 20 seconds before timing out.
+			const int CoinjoinProcessTimeout = 40_000;
+			var transactionCompleted = new TaskCompletionSource<Transaction>();
+			using var cts = new CancellationTokenSource(CoinjoinProcessTimeout);
+			cts.Token.Register(() => transactionCompleted.TrySetCanceled(), useSynchronizationContext: false);
+
+			// Create a key manager and use it to create two fake coins.
+			var keyManager = KeyManager.CreateNew(out var _, password: "");
+			keyManager.AssertCleanKeysIndexed();
+			var coins = keyManager.GetKeys()
+				.Take(InputCount)
+				.Select(x => new Coin(
+					BitcoinFactory.CreateOutPoint(),
+					new TxOut(Money.Coins(1), x.P2wpkhScript)))
+				.ToArray();
+
+			var httpClient = _apiApplicationFactory.WithWebHostBuilder(builder =>
+			{
+				builder.ConfigureServices(services =>
+				{
+					var rpc = BitcoinFactory.GetMockMinimalRpc();
+
+					// Make the coordinator to believe that those two coins are real and
+					// that they exist in the blockchain with many confirmations.
```

Comment:
just wanted to say that I really appreciate these kinds of comments in test code, it saves a lot of effort for the next person